### DISCUSSION TITLE
ViomiV7 (STYJ02YM): Consumables not working correctly

### DISCUSSION THREAD

**Discussion by nawuko** [Hypfer/Valetudo#754] [Q&A / Support]
Hey, i just bought the V7 a few days ago, running latest Valetudo with MQTT and HA together, awesome stuff so far.

Sadly consumables always show the "full" remaining time, inside Valetudo itself and MQTT

Looking at log data, it looks like the return value of `get_consumables` has changed, it no longer uses minutes it instead uses hours in use and the prop.hypa_hours shows the remaining use time. (FW: 3.5.3_0047)

My suggestion would be to implement a proper way to use the new props `prop.hypa_hours` instead of hard-coding the ("total lifetime" - used time) implementation that currently is used with `get_consumables`

I'm able to make a PR but wanted to ask first what should be a proper way of implementing this?

Currently i use this: https://gist.github.com/nawuko/9a3b1ae02f07763760bb7ba285e1921f, which works correctly in Valetudo and Home Assistant/MQTT

**Comment by 8times4** [comment]
@Hypfer any chance to get this into Valetudo? Looks like a nice to have feature. Thanks!

**Comment by depau** [comment]
> Currently i use this: https://gist.github.com/nawuko/9a3b1ae02f07763760bb7ba285e1921f, which works correctly in Valetudo and Home Assistant/MQTT

LGTM, you can send a pull request with this code.
